spark/src/test/java/org/apache/iceberg/spark/source/TestDataFrameWrites.java

```diff
@@ -233,17 +234,14 @@ private void writeData(Iterable<Record> records, Schema schema, String location)
 
   @Test
   public void testNullableWithWriteOption() throws IOException {
+    Assume.assumeTrue("Spark 3.0 rejects writing nulls to a required column", spark.version().startsWith("2"));
```

Comment:
Does this mean we will no longer support `check-nullability` option in Spark 3?

Reply:
I see `check-nullability` in the Spark 3 code, though.

Reply:
We can support it for `DataFrame` operations by returning a table schema with all optional columns when loading the table. But since that's more work to produce an all-optional schema, I think it should be done in a follow-up.